File: Data Structures & Algorithms/anagram-groups/submission-2.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        const map = new Map();

        for (const str of strs) {
            const charArray = new Array(26).fill(0);

            for (const c of str) {
                charArray[c.charCodeAt(0)-'a'.charCodeAt(0)]++;
            }
            if (map.has(charArray.toString())) {
                map.get(charArray.toString()).push(str)
            } else {
                map.set(charArray.toString(), [str])
            }
        }

        let res = [];

        map.forEach((value, key) => {
            let cur = [];

            for(const val of value) {
                cur.push(val);
            }
            res.push(cur);
        })
        return res;

    }
}

File: Data Structures & Algorithms/add-binary/submission-2.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        i, j = len(a) - 1, len(b) - 1
        res = ""
        carry = False
        while i >= 0 and j >= 0:
            if a[i] == "1" and b[j] == "1" and carry:
                res = "1" + res
            elif a[i] == "1" and b[j] == "1" or ((a[i] == "1" or b[j] == "1") and carry):
                res = "0" + res
                carry = True
            elif a[i] == "1" or b[j] == "1" or carry:
                res = "1" + res
                carry = False
            else:
                res = "0" + res
                carry = False
            i -= 1
            j -= 1
        
        while i >= 0:
            if carry and a[i] == "1":
                res = "0" + res
            elif a[i] == "1" or carry:
                res = "1" + res
                carry = False
            else:
                res = "0" + res
                carry = False
            i -= 1
        
        while j >= 0:
            if carry and b[j]  == "1":
                res = "0" + res
            elif b[j] == "1" or carry:
                res = "1" + res
                carry = False
            else:
                res = "0" + res
                carry = False
            j -= 1
        
        if carry:
            res = "1" + res
        return res
        
        
        